spinalcordtoolbox/deepseg/core.py

```diff
@@ -0,0 +1,144 @@
+# coding: utf-8
+"""
+Interface API for the deepseg module, which performs segmentation using deep learning with the ivadomed package.
+"""
+
+
+import logging
+import numpy as np
+import nibabel as nib
+import ivadomed as imed
+import ivadomed.utils
+import ivadomed.postprocessing
+
+import spinalcordtoolbox as sct
+import spinalcordtoolbox.deepseg.models
+
+
+logger = logging.getLogger(__name__)
+
+# Default values if not asked during CLI call and if not present in json metadata.
+DEFAULT_THRESHOLD = 0.9
+DEFAULT_KEEP_LARGEST_OBJECT = True
+DEFAULT_FILL_HOLES = True
+
+
+class ParamDeepseg:
+    """
+    Parameters for deepseg module.
+    """
+    def __init__(self):
+        self.threshold = None
+        self.keep_largest_object = None
+        self.fill_holes = None
+        self.output_suffix = '_seg'
+        self.remove_temp_files = 1
+        self.verbose = 1
+        # TODO: add threshold, keep_big_obj
+
+
+class PostProcessing:
+    """
+    Deals with post-processing of the segmentation. Consider param (i.e. user's flag) with more priority than
+    metadata (i.e. from model's json file).
+    """
+    def __init__(self, param, metadata):
+        """
+        :param param: class ParamDeepseg: Defined by user's parameter
+        :param metadata: dict: From model's json metadata
+        """
+        self.param = param
+        self.metadata = metadata
+
+    def threshold(self, nii_seg):
+        """
+        Threshold the prediction. For no prediction, set 'threshold' to 0.
+        """
+        if self.param.threshold:
```

Comment:
You can use `threshold = None` to mean "no threshold". That would be more obvious and uniform with the other settings. It's a remnant of C's limited type system that people repurposed the meanings of specific values.

Reply:
no more relevant with current implementation (ce8a184ec687967d3b0186f21614d6716d797d8a)